### ISSUE TITLE
Crash when validating VST3 plugins

### ISSUE THREAD

**Issue by benthevining** [Tracktion/pluginval#63]
I originally posted this as a comment on [this issue](https://github.com/Tracktion/pluginval/issues/62), but it was since closed so I wanted to open a new issue. Apologies if this is unappreciated.

When testing a VST3 plugin from the command line, we occasionally get a crash with this stack trace:

```
Starting test: pluginval / Editor Automation...

*** FAILED: VALIDATION CRASHED WHILST VALIDATING /Library/Audio/Plug-Ins/VST3/CenterOne.vst3
0   pluginval                           0x00000001008721a0 _ZN4juce11SystemStats17getStackBacktraceEv + 64
1   pluginval                           0x00000001006bd9e6 _ZN12_GLOBAL__N_119getCrashLogContentsEv + 38
2   pluginval                           0x00000001006bd98b _Z11getCrashLogv + 107
3   pluginval                           0x00000001006b8122 _ZN20CommandLineValidator14connectionLostEv + 50
4   pluginval                           0x00000001006d26e8 _ZNSt3__110__function6__funcIZN9Validator16ensureConnectionEvE3$_1NS_9allocatorIS3_EEFvvEEclEv + 56
5   pluginval                           0x00000001006d2444 _ZN22ValidatorParentProcess20handleConnectionLostEv + 68
6   pluginval                           0x000000010090f379 _ZN4juce12MessageQueue21runLoopSourceCallbackEPv + 57
7   CoreFoundation                      0x00007ff80880f35a __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 17
8   CoreFoundation                      0x00007ff80880f2c2 __CFRunLoopDoSource0 + 180
9   CoreFoundation                      0x00007ff80880f040 __CFRunLoopDoSources0 + 242
10  CoreFoundation                      0x00007ff80880da50 __CFRunLoopRun + 892
11  CoreFoundation                      0x00007ff80880d014 CFRunLoopRunSpecific + 562
12  HIToolbox                           0x00007ff8119215e6 RunCurrentEventLoopInMode + 292
13  HIToolbox                           0x00007ff81192134a ReceiveNextEventCommon + 594
14  HIToolbox                           0x00007ff8119210e5 _BlockUntilNextEventMatchingListInModeWithFilter + 70
15  AppKit                              0x00007ff80b1831fd _DPSNextEvent + 927
16  AppKit                              0x00007ff80b1818ba -[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 1394
17  AppKit                              0x00007ff80b173f69 -[NSApplication run] + 586
18  pluginval                           0x0000000100903274 _ZN4juce19JUCEApplicationBase4mainEv + 164
19  pluginval                           0x0000000100903183 _ZN4juce19JUCEApplicationBase4mainEiPPKc + 83
20  dyld                                0x000000020129851e start + 462
21  dyld                                0x0000000201293000 dyld + 0

Binary Images:
0x1006ad000 pluginval
0x7ff808790000 CoreFoundation
0x7ff8118f3000 HIToolbox
0x7ff80b144000 AppKit

libc++abi: Pure virtual function called!
```

The failure appears to only happen from the command line, I've never had it happen when running from the GUI app (though this may be coincidental).

When I run `pluginval --validate <plugin> --strictnessLevel 10` all tests pass, but when I add any of the randomize, repeat, samplerates, or blocksize options I get this crash.

Does this appear to be a bug in pluginval itself or in our plugin code? If it's a bug in our code, does anyone have any suggestions for where to start looking?

**Comment by drowaudio** [comment]
Is this fixed now with the 1.0.0 release?

**Comment by benthevining** [comment]
Seems to be, yes. Thank you!
